src/methods.ts

```diff
@@ -91,6 +91,9 @@ export interface MessageAttachment {
     type: string;
     text?: string;
   }[];
+  attachment_type?: 'default';
+  callback_id?: string;
+  mrkdwn_in?: ('pretext' | 'text' | 'fields')[];
```

Comment:
this works and its right, but i was a little confused by the syntax. would you mind making a named string-based enum type and then setting this type to an array of that?

Reply:
I would prefer to use a union type instead of enum, because declare it as an enum type, e.g.:
```
export enum MrkdwnIn {
  Pretext: 'pretext',
  Text: 'text',
  Fields: 'fields'
}

MessageAttachment {
  //...
  mrkdwn_in?: MrkdwnIn[];
}
```
it won't be possible to define mesages in this "plain" way
```
const attachment: MessageAttachment = {
  mrkdwn_in: ['pretext']
}
```
and forces a user to use this style:
```
const attachment: MessageAttachment = {
  mrkdwn_in: ['MrkdownIn.pretext']
}
```

Reply:
😮 i guess i never really gave any thought to the intricate differences between a union type and an enum. i always thought that enums were a "more powerful" union type because TypeScript would generate a symbol with the actual map, but didn't realize the compiler would complain when you use the literal values like in your example. [i tried it out here](http://www.typescriptlang.org/play/#src=enum%20MrkdwnInOne%20%7B%0D%0A%20%20Pretext%20%3D%20'pretext'%2C%0D%0A%20%20Text%20%3D%20'text'%2C%0D%0A%20%20Fields%20%3D%20'fields'%0D%0A%7D%0D%0A%0D%0Ainterface%20MessageAttachmentOne%20%7B%0D%0A%20%20%2F%2F...%0D%0A%20%20mrkdwn_in%3F%3A%20MrkdwnInOne%5B%5D%3B%0D%0A%7D%0D%0A%0D%0Aconst%20myAttachment%3A%20MessageAttachmentOne%20%3D%20%7B%0D%0A%20%20%20%20mrkdwn_in%3A%20%5BMrkdwnInOne.Pretext%5D%0D%0A%7D%0D%0A%0D%0Atype%20MrkdwnInTwo%20%3D%20'pretext'%20%7C%20'text'%20%7C%20'fields'%3B%0D%0A%0D%0Ainterface%20MessageAttachmentTwo%20%7B%0D%0A%20%20%2F%2F...%0D%0A%20%20mrkdwn_in%3F%3A%20MrkdwnInTwo%5B%5D%3B%0D%0A%7D%0D%0A%0D%0Aconst%20anotherAttachment%3A%20MessageAttachmentTwo%20%3D%20%7B%0D%0A%20%20%20%20mrkdwn_in%3A%20%5B'text'%2C%20'pretext'%5D%0D%0A%7D%3Bt).

in light of this fact, i agree that the union type is a better solution. thanks!